src/main/kotlin/kweb/html/events/OnReceiver.kt

```diff
@@ -25,9 +26,15 @@ class OnReceiver<T : EventGenerator<T>>(internal val source: T, private val retr
         val eventPropertyNames = memberProperties(U::class)
 
         val serializer = serializer<U>()
+        source.browser.kweb.isCatchingOutbound()
         return event(eventName, eventPropertyNames) { propertiesAsElement ->
             val props = Json.decodeFromJsonElement(serializer, propertiesAsElement)
             try {
+                if (source.browser.kweb.isCatchingOutbound() != null) {
+                    source.browser.kweb.catchOutbound(Kweb.CatcherType.BATCH) {
+                        callback(props)
+                    }
+                }
```

Comment:
Shouldn't the next line be inside an `else {}` block? Seems like this might call `callback` twice.

Reply:
That's what I get for editing this after midnight. It's strange that this didn't break anything, but yes, this should be in an else block.